File: src/main/java/poker/ignition/Seat.java
package poker.ignition;

import com.google.common.base.MoreObjects;
import poker.domain.HoleCards;

import java.util.Objects;

public class Seat {

    private final int number;
    private final Position position;
    private final HoleCards holeCards;
    private final boolean me;
    private long stack;
    private final long profitLoss;
    private final long cashDeposit;

    public Seat(int number, Position position, HoleCards holeCards,
                boolean me, long stack, long profitLoss, long cashDeposit) {
        this.number = number;
        this.position = position;
        this.holeCards = holeCards;
        this.me = me;
        this.stack = stack;
        this.profitLoss = profitLoss;
        this.cashDeposit = cashDeposit;
    }

    public int getNumber() {
        return number;
    }

    public Position getPosition() {
        return position;
    }

    public HoleCards getHoleCards() {
        return holeCards;
    }

    public boolean isMe() {
        return me;
    }

    public long getStack() {
        return stack;
    }

    public long getProfitLoss() {
        return profitLoss;
    }

    public long getCashDeposit() {
        return cashDeposit;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Seat seat = (Seat) o;
        return number == seat.number &&
                me == seat.me &&
                stack == seat.stack &&
                profitLoss == seat.profitLoss &&
                cashDeposit == seat.cashDeposit &&
                position == seat.position &&
                Objects.equals(holeCards, seat.holeCards);
    }

    @Override
    public int hashCode() {
        return Objects.hash(number, position, holeCards, me, stack, profitLoss, cashDeposit);
    }

    @Override
    public String toString() {
        return MoreObjects.toStringHelper(this)
                .add("number", number)
                .add("position", position)
                .add("holeCards", holeCards)
                .add("me", me)
                .add("stack", stack)
                .add("profitLoss", profitLoss)
                .add("cashDeposit", cashDeposit)
                .toString();
    }
}

File: src/main/kotlin/poker/ignition/Seat.kt
package poker.ignition

class Seat(val number: Int, val position: Position, val me: Boolean, val stack: Long) {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Seat

        if (number != other.number) return false
        if (position != other.position) return false
        if (me != other.me) return false
        if (stack != other.stack) return false

        return true
    }

    override fun hashCode(): Int {
        var result = number
        result = 31 * result + position.hashCode()
        result = 31 * result + me.hashCode()
        result = 31 * result + stack.hashCode()
        return result
    }

    override fun toString(): String {
        return "Seat(number=$number, position=$position, me=$me, stack=$stack)"
    }


}